courier/http_channel.go

```diff
@@ -109,9 +112,13 @@ func (c *httpChannel) Dispatch(ctx context.Context, msg Message) (err error) {
 		"unable to dispatch mail delivery because upstream server replied with status code %d",
 		res.StatusCode,
 	)
+
+	body, _ := io.ReadAll(res.Body)
```

Comment:
res.body has ben read already in L96, this will fail with an empty body

Reply:
This could be written more simply as

```go
	defer res.Body.Close()
	body := make([]byte, 1024)
	n, _ :=res.Body.Read(body)
	body = body[:n]
```

Reply:
Ah, makes sense!